tools/importer/importer.py

```diff
@@ -0,0 +1,254 @@
+    if name in output:
+        return True
+    return False
+
+def branch_checkout(name):
+    """
+    Checkout the required branch
+    Args:
+    name - branch name
+    """
+    cmd = "git checkout " + name
+    run_cmd_with_output(cmd, exit_on_failure=False)
+
+def get_last_cherry_pick_sha(branch):
+    """
+    SHA of last cherry pick commit is returned. SHA should be added to all
+    cherry-pick commits with -x option.
+
+    Args:
+    branch - Hash to be verified.
+    Returns - SHA if found, else None
+    """
+    cmd = "git checkout " + branch
+    run_cmd_with_output(cmd, exit_on_failure=False)
+
+    SHA = None
+    get_commit = "git log -n 1"
+    _, output = run_cmd_with_output(get_commit, exit_on_failure=True)
+    lines = output.split('\n')
+    for line in lines:
+        if 'cherry picked from' in line:
+            SHA = line.split(' ')[-1]
+            SHA = SHA[:-1]
+    return SHA
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description=__doc__,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('-l', '--log-level',
+                        help="Level for providing logging output",
+                        default='INFO')
+    parser.add_argument('-r', '--repo-path',
+                        help="Git Repository to be imported",
+                        default=None,
+                        required=True)
+    parser.add_argument('-c', '--config-file',
+                        help="Configuration file",
+                        default=None,
+                        required=True)
+    args = parser.parse_args()
+    level = getattr(logging, args.log_level.upper())
+
+    # Set logging level
+    logging.basicConfig(level=level)
+    rel_log = logging.getLogger("Importer")
+
+    if (args.repo_path is None) or (args.config_file is None):
+        rel_log.error("Repository path and config file required as input. Use \"--help\" for more info.")
+        exit(1)
+
+    json_file = os.path.abspath(args.config_file)
+    if not os.path.isfile(json_file):
+        rel_log.error("%s not found.", args.config_file)
+        exit(1)
+
+    repo = os.path.abspath(args.repo_path)
+    if not os.path.exists(repo):
+        rel_log.error("%s not found.", args.repo_path)
+        exit(1)
+
+    sha = get_curr_sha(repo)
+    if not sha:
+        rel_log.error("Could not obtain latest SHA")
+        exit(1)
+    rel_log.info("%s SHA = %s", os.path.basename(repo), sha)
+
+    branch = 'feature_' + os.path.basename(repo) + '_' + sha
+    commit_msg = "[" + os.path.basename(repo) + "]" + ": Updated to " + sha
+
+    # Read configuration data
+    with open(json_file, 'r') as config:
+        json_data = json.load(config)
+
+    '''
+    Check if branch exists already, in case branch is present
+    we will skip all file transfer and merge operations and will
+    jump to cherry-pick
+    '''
+    if check_branch(branch):
```

Comment:
super naming nit: I have no idea what check branch is checking for from it's name. From the definition, and the logging below, I was able to gather that it checks for the presence of a branch so I think it would be better named `branch_exists`.